<?xml version="1.0" encoding="UTF-8"?>

<launch>

	<arg name="launch_prefix" default="" doc="gdb -ex run --args"/>
	<arg name="launch_rviz" default="false" />
	<arg name="cloud_topic" default="/ouster_points_self_filtered" />
	<arg name="image_topic" default="/camMainView/image_raw" />
	<arg name="use_rgb_from_camera" default="true" />
	<arg name="parameter_file_path" default="$(find open3d_slam)/param/params_ouster_os0_128.yaml"/>
	<arg name="num_accumulated_range_data" default="1"/>
	<arg name="is_read_from_rosbag" default="false"/>
	<arg name="rosbag_filepath" default=""/>
	<arg name="use_sim_time" default="false"/>

	<param name="/use_sim_time" value="$(arg use_sim_time)" />
	
	<node name="mapping_node" pkg="open3d_slam"
		type="mapping_node" output="screen" launch-prefix="$(arg launch_prefix)">

		<param name="cloud_topic" type="string"
			value="$(arg cloud_topic)" />

		<param name="image_topic" type="string"
			value="$(arg image_topic)" />

		<param name="use_rgb_from_camera" type="bool"
			value="$(arg use_rgb_from_camera)" />
			
		<param name="parameter_file_path" type="string"
			value="$(arg parameter_file_path)" />
			
		<param name="num_accumulated_range_data" value="$(arg num_accumulated_range_data)"/>
		<param name="is_read_from_rosbag" value="$(arg is_read_from_rosbag)"/>
		<param name="rosbag_filepath" value="$(arg rosbag_filepath)"/>

	</node>


	<include
		file="$(find open3d_slam)/launch/vis.launch"
		if="$(arg launch_rviz)">
	</include>

</launch>